File: android/src/lollipop/java/de/idrinth/habitevaluator/android/NotificationHelper.java
package de.idrinth.habitevaluator.android;

import android.content.Context;

/**
 * Lollipop flavor notification helper (API 21–25).
 * Notification channels do not exist before API 26, so channel
 * creation is a no-op. Devices running API 26+ use the oreo flavor.
 */
public final class NotificationHelper {

    public static final String CHANNEL_ID = "habit_evaluator_reminders";

    private NotificationHelper() {
    }

    public static void ensureNotificationChannel(Context context) {
        // No-op: notification channels were introduced in API 26 (Oreo).
    }

    public static NotificationPermissionHandler permissionHandler() {
        return new PreApi33PermissionHandler();
    }
}

File: android/src/lollipop/kotlin/de/idrinth/habitevaluator/android/NotificationHelper.kt
package de.idrinth.habitevaluator.android

import android.content.Context

object NotificationHelper {
    const val CHANNEL_ID = "habit_evaluator_reminders"
    const val CHANNEL_ID_INPUT_REMINDERS = "habit_evaluator_input_reminders"
    const val CHANNEL_ID_EMOTION = "habit_evaluator_emotion"
    const val CHANNEL_ID_DAY_PLANNER = "habit_evaluator_day_planner"

    fun ensureNotificationChannel(context: Context) {
        // No-op: notification channels and groups were introduced in API 26 (Oreo).
    }

    fun permissionHandler(): NotificationPermissionHandler = PreApi33PermissionHandler()
}
